source/java/src/org/lucee/extension/cache/eh/util/EHCacheClassLoader.java

```diff
@@ -0,0 +1,135 @@
+package org.lucee.extension.cache.eh.util;
+
+import java.lang.reflect.Method;
+
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+
+import lucee.loader.engine.CFMLEngine;
+import lucee.loader.engine.CFMLEngineFactory;
+import lucee.runtime.config.Config;
+
+import lucee.commons.io.log.Log;
+
+
+/**
+ * A special class loader that first attempts to load class from the Lucee Environment,
+ * but that can fail when RMI classes are requested via a spawned thread, so when
+ * the class cannot be located in the Lucee Environment, it searches all the Felix
+ * bundles for the requested class. This will prevent issues with issues trying
+ * to load classes like net.sf.ehcache.distribution.RMICachePeer_Stub. 
+ *
+ * @author Dan G. Switzer, II
+ * @version 1.0
+ */
+public class EHCacheClassLoader extends ClassLoader {
+	private Config config;
+	private ClassLoader LuceeEnvClassLoader;
+
+	public EHCacheClassLoader(Config config) {
+		this.config = config;
+
+		try {
+			Method m = config.getClass().getMethod("getClassLoaderEnv", new Class[0]);
+			this.LuceeEnvClassLoader = (ClassLoader) m.invoke(config, new Object[0]);
+		}
+		catch (Exception e) {
+			// do nothing
+		}
+	}
+
+	public CFMLEngine getCFMLEngine(){
+		try {
+			Class<?> clazz = CFMLEngineFactory.getInstance().getClassUtil().loadClass("lucee.runtime.config.ConfigWebUtil");
+			Method m = clazz.getMethod("getEngine", new Class[] { Config.class });
+			return (CFMLEngine) m.invoke(null, this.config);
+		} catch (Exception e) {
+			e.printStackTrace();
```

Comment:
we should not simply write this to the console, throw it or log it.

Reply:
Sorry, left over debug code that I didn't remove. I've replaced it with a comment that it's safe to ignore the exception.